lucille-core/src/main/java/com/kmwllc/lucille/stage/PromptOllama.java

```diff
@@ -0,0 +1,140 @@
+ *  asking for a single piece of information (like a summary).
+ *
+ *  fields (list of Strings, Optional): The fields in the document you want to be sent to the LLM. If the list is empty or not specified,
+ *  defaults to sending the entire Document to the LLM for enriching.
+ *
+ *  requireJSON (Boolean, Optional): Whether you are requiring & expecting the LLM to output a JSON-only response. When true,
+ *  Lucille will throw an Exception upon receiving a non-JSON response from the LLM. When false, Lucille will place the response's
+ *  raw contents into the "ollamaResponse" field. Defaults to false.
+ */
+public class PromptOllama extends Stage {
+
+  private static final Logger log = LoggerFactory.getLogger(PromptOllama.class);
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  private final String hostURL;
+  private final String modelName;
+  private final Long timeout;
+
+  private final String systemPrompt;
+  private final List<String> fields;
+  private final boolean requireJSON;
+
+  private OllamaAPI ollamaAPI;
+  private OllamaChatRequestBuilder chatBuilder;
+
+  public PromptOllama(Config config) {
+    super(config, new StageSpec()
+        .withRequiredProperties("hostURL", "modelName", "systemPrompt")
+        .withOptionalProperties("timeout", "fields", "requireJSON"));
+
+    this.hostURL = config.getString("hostURL");
+    this.modelName = config.getString("modelName");
+    this.timeout = ConfigUtils.getOrDefault(config, "timeout", null);
+
+    this.systemPrompt = config.getString("systemPrompt");
+    this.fields = ConfigUtils.getOrDefault(config, "fields", List.of());
+    this.requireJSON = ConfigUtils.getOrDefault(config, "requireJSON", false);
+  }
+
+  @Override
+  public void start() throws StageException {
+    this.ollamaAPI = new OllamaAPI(hostURL);
+    this.chatBuilder = OllamaChatRequestBuilder
+        .getInstance(modelName)
+        .withMessage(OllamaChatMessageRole.SYSTEM, systemPrompt);
+
+    if (timeout != null) {
+      ollamaAPI.setRequestTimeoutSeconds(timeout);
+    }
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+    OllamaChatRequest request = createRequestWithSpecifiedFields(doc);
+
+    OllamaChatResult chatResult;
+    try {
+      chatResult = ollamaAPI.chat(request);
+    } catch (Exception e) {
+      throw new StageException("Error communicating with Ollama server.", e);
+    }
+
+    // Always try to get JSON from the response, even if requireJSON is false.
+    try {
+      JsonNode node = mapper.readTree(chatResult.getResponseModel().getMessage().getContent());
+      // put all the fields from the JSON onto the Lucille Document.
+      node.fields().forEachRemaining(entry -> doc.setField(entry.getKey(), entry.getValue()));
+    } catch (JsonProcessingException e) {
+      if (requireJSON) {
+        throw new StageException("Error getting JSON from response", e);
```

Comment:
I think we should specify in this log that `requireJSON` was set to true and we did not get a valid JSON response